test/module-test.cc

```diff
@@ -55,11 +76,551 @@ TEST(module_test, detail_namespace) {
+  EXPECT_THROW(throw fmt::format_error("oops"), std::exception);
+  EXPECT_THROW(throw fmt::vsystem_error(0, "{}", store),
+               std::system_error);
+  EXPECT_THROW(throw fmt::system_error(0, "{}", 42),
+               std::system_error);
+
+  fmt::memory_buffer buffer;
+  fmt::format_system_error(buffer, 0, "oops");
+  auto oops = to_string(buffer);
+  EXPECT_TRUE(oops.size() > 0);
+  EXPECT_WRITE(stderr, fmt::report_system_error(0, "oops"), oops + '\n');
+
+#ifdef _WIN32
+  EXPECT_THROW(throw fmt::vwindows_error(0, "{}", store),
+               std::system_error);
+  EXPECT_THROW(throw fmt::windows_error(0, "{}", 42),
+               std::system_error);
+  output_redirect redirect(stderr);
+  fmt::report_windows_error(0, "oops");
+  EXPECT_TRUE(redirect.restore_and_read().size() > 0);
+#endif
+}
+
+TEST(module_test, error_code) {
+  EXPECT_EQ("generic:42",
+            fmt::format("{0}",
+                        std::error_code(42, std::generic_category())));
+  EXPECT_EQ("system:42",
+            fmt::format("{0}",
+                        std::error_code(42, fmt::system_category())));
+}
+
+TEST(module_test, format_int) {
+  fmt::format_int sanswer(42);
+  EXPECT_EQ("42", fmt::string_view(sanswer.data(), sanswer.size()));
+  fmt::format_int uanswer(42u);
+  EXPECT_EQ("42", fmt::string_view(uanswer.data(), uanswer.size()));
+}
+
+struct test_formatter : fmt::formatter<char> {
+  bool check() { return true; }
+};
+
+struct test_dynamic_formatter : fmt::dynamic_formatter<> {
+  bool check() { return true; }
+};
+
+TEST(module_test, formatter) {
+  EXPECT_TRUE(test_formatter{}.check());
+  EXPECT_TRUE(test_dynamic_formatter{}.check());
+}
+
+TEST(module_test, join) {
+  int arr[3] = {1, 2, 3};
+  std::vector<double> vec{1.0, 2.0, 3.0};
+  EXPECT_EQ("1, 2, 3", to_string(fmt::join(arr + 0, arr + 3, ", ")));
+  EXPECT_EQ("1, 2, 3", to_string(fmt::join(arr, ", ")));
+  EXPECT_EQ("1, 2, 3", to_string(fmt::join(vec.begin(), vec.end(), ", ")));
+  EXPECT_EQ("1, 2, 3", to_string(fmt::join(vec, ", ")));
+}
+
+TEST(module_test, time) {
+  auto time_now = std::time(nullptr);
+  EXPECT_TRUE(fmt::localtime(time_now).tm_year > 120);
+  EXPECT_TRUE(fmt::gmtime(time_now).tm_year > 120);
+  auto chrono_now = std::chrono::system_clock::now();
+  EXPECT_TRUE(fmt::localtime(chrono_now).tm_year > 120);
+  EXPECT_TRUE(fmt::gmtime(chrono_now).tm_year > 120);
+}
+
+TEST(module_test, time_point) {
+  auto now = std::chrono::system_clock::now();
+  std::string_view past("2021-05-20 10:30:15");
+  EXPECT_TRUE(past < fmt::format("{:%Y-%m-%d %H:%M:%S}", now));
+}
+
+TEST(module_test, time_duration) {
+  EXPECT_EQ("42s", fmt::format("{}", std::chrono::seconds{42}));
+  using us = std::chrono::duration<double, std::micro>;
+  EXPECT_EQ("4.2µs", fmt::format("{:3.1}", us{4.234}));
+  const auto de = std::locale("de_DE");
+  EXPECT_EQ("4.2µs", fmt::format(de, "{:L}", us{4.2}));
+}
+
+TEST(module_test, weekday) {
+  const auto de = std::locale("de_DE");
```

Comment:
This may throw if the locale is not available.

Reply:
Yeah ...
Luckily we don't care here.

Reply:
Why don't we care? Since this is a non-functional test I suggest using the `std::locale::classic()` which is always available.

Reply:
We don't care for the same reason why you suggest using the classic locale: it just has to look up the correct overloads. If you prefer `std::locale::classic()` then I'll change it accordingly.

Reply:
> it just has to look up the correct overloads.

It would enough if we didn't actually run this code, only compile. But my understanding is that we eventually want to run this code.

> If you prefer std::locale::classic() then I'll change it accordingly.

Please do.